pypeit/scripts/coadd_2dspec.py

```diff
@@ -78,121 +51,109 @@ def get_parser(cls, width=None):
+#        qa_path = coadd_qa_dir / 'PNGs'
+        par['rdx']['qadir'] += '_coadd'
+        qa_path = pypeit_scidir.parent / par['rdx']['qadir'] / 'PNGs'
+        if not qa_path.exists():
+            qa_path.mkdir(parents=True)
+
+        # Get the output basename
         head2d = fits.getheader(spec2d_files[0])
         if args.basename is None:
-            #TODO Fix this, currently does not work if target names have - or _
-            filename_first = os.path.basename(spec2d_files[0])
-            filename_last = os.path.basename(spec2d_files[-1])
-            prefix_first = (filename_first.split('_')[1]).split('-')[0]
-            prefix_last = (filename_last.split('_')[1]).split('-')[0]
-            objname = (filename_first.split('-')[1]).split('_')[0]
-            basename = '{:s}-{:s}-{:s}'.format(prefix_first,prefix_last,objname)
+            lasthdr = fits.getheader(spec2d_files[-1])
+            if 'FILENAME' not in head2d:
+                msgs.error(f'Missing FILENAME keyword in {spec2d_files[0]}.  Set the basename '
+                           'using the command-line option.')
+            if 'FILENAME' not in lasthdr:
+                msgs.error(f'Missing FILENAME keyword in {spec2d_files[-1]}.  Set the basename '
+                           'using the command-line option.')
+            if 'TARGET' not in head2d:
+                msgs.error(f'Missing TARGET keyword in {spec2d_files[0]}.  Set the basename '
+                           'using the command-line option.')
+            basename = f"{io.remove_suffix(head2d['FILENAME'])}-" \
+                       f"{io.remove_suffix(lasthdr['FILENAME'])}-{head2d['TARGET']}"
         else:
             basename = args.basename
 
-
-        # TODO Heliocentric for coadd2d needs to be thought through. Currently turning it off.
-        parset['calibrations']['wavelengths']['refframe'] = 'observed'
-        # TODO Flexure correction for coadd2d needs to be thought through. Currently turning it off.
-        parset['flexure']['spec_method'] = 'skip'
-        # TODO This is currently the default for 2d coadds, but we need a way to toggle it on/off
-        parset['reduce']['findobj']['skip_skysub'] = True
         # Write the par to disk
-        par_outfile = basename+'_coadd2d.par'
-        print("Writing the parameters to {}".format(par_outfile))
-        parset.to_config(par_outfile, exclude_defaults=True, include_descr=False)
+        par_outfile = coadd_scidir.parent / f'{basename}_coadd2d.par'
+        print(f'Writing full parameter set to {par_outfile}.')
+        par.to_config(par_outfile, exclude_defaults=True, include_descr=False)
 
         # Now run the coadds
-
-        skysub_mode = head2d['SKYSUB']
-        findobj_mode = head2d['FINDOBJ']
-        bkg_redux = True if 'DIFF' in skysub_mode else False
-        find_negative = True if 'NEG' in findobj_mode else False
+        bkg_redux = head2d['SKYSUB'] == 'DIFF'
+        find_negative = head2d['FINDOBJ'] == 'NEG'
```

Comment:
I found a small bug. This should be `find_negative = 'NEG' in head2d['FINDOBJ']`

Reply:
Why is it a bug, is this not equivalent but more robust?

Reply:
`head2d['FINDOBJ']` is not equal to 'NEG', at least in the data I tested. It was equal to 'POS_NEG' (or something like that). I don't know where this header key is created and what are the options, this is why I suggested `'NEG' in head2d['FINDOBJ']`. 
As it is now, the code does not find the negative traces.

Reply:
Thanks for catching this!  I've changed the relevant line to `find_negative = head2d['FINDOBJ'] == 'POS_NEG'`.